.ci/gitlab-ci.yml

```
@@ -0,0 +1,48 @@
+# Gitlab configuraiton for spack/spack
+
+stages:
+  - packages
+
+variables:
+  SPACK_PACKAGES_CHECKOUT_VERSION: refs/pull/342/head
+
+.clone_packages: &clone_packages
+  - mkdir -p ${REPO_DESTINATION}
+  - cd ${REPO_DESTINATION}
+  - git init
+  - git remote add origin https://github.com/spack/spack-packages.git
+  - git fetch --depth 1 origin ${SPACK_PACKAGES_CHECKOUT_VERSION}
+  - git checkout FETCH_HEAD
+  - cd -
+
+dotenv:
+  stage: .pre
+  image: ghcr.io/spack/e4s-ubuntu-18.04:v2021-10-18
+  tags: [ spack, service ]
+  script:
+    - export REPO_DESTINATION=etc/spack-packages
+    - *clone_packages
+    - repo_commit=$(git -C ${REPO_DESTINATION} rev-parse FETCH_HEAD)
+    # When running CI from spack/spack all packages are untouched
+    # so skip pruning untouched. Deferred pipelines _should_ handle
+    # most of the cases where this type of pruning would need to be
+    # applied.
+    - echo "SPACK_CHECKOUT_VERSION=${repo_commit}" >> ${CI_PROJECT_DIR}/env
+    - echo "SPACK_CHECKOUT_REPO=spack/spack-packages" >> ${CI_PROJECT_DIR}/env
```

```suggestion
    - echo "SPACK_PACKAGES_CHECKOUT_VERSION=${repo_commit}" >> ${CI_PROJECT_DIR}/env
    - echo "SPACK_PACKAGES_CHECKOUT_REPO=spack/spack-packages" >> ${CI_PROJECT_DIR}/env
```
I'm also not sure I understand what the second line here is doing -- we didn't clone it into spack/spack-packages.

---

There are two things happening here. The first is cloning packages and getting the stacks/CI config from that project. The second is overriding variables set in packages CI to work from core. This line here is doing the latter part, it makes a little more sense when you look at the changes in the spack/spack-packages PR which uses these variables.

---

Trying to be more clear here.

There are two repos we must clone every time we run CI: `spack/spack` and `spack/spack-packages`.

`SPACK_CHECKOUT_*` is used to handle the repo we must checkout _manually_. In the context of `spack/spack` that repo is `spack/spack-packages`. In the context of `spack/spack-packages` that repo is `spack/spack`

The reason we reuse the name here is so that I don't have to duplicate all of the CI configuration, we can just reuse it from `spack/spack-packages` directly. We handle configuring the roots of the two projects in the `spack/spack-packages` PR with the `write_dotenv.py` script which looks at the value of `CI_PROJECT_NAME` to determine where we are running CI from.